src/main.js

```diff
@@ -170,62 +190,41 @@ async function startFromFile(configPath, rulesPath) {
   }
 
-  const rml = await yarrrmlToRml(yarrrml);
+  rml = await yarrrmlToRml(yarrrml);
 
   await setupAuth();
 
   // Cold start
   ymlContentToConfig(configYml);
   const {results, keys} = await queryResource(config, true);
-  if (Object.keys(results).length === 0){
+  if (Object.keys(results).length === 0) {
     console.error('Failed cold start, no data collected from pod');
     return;
   }
   config.keys = [...keys];
   const arrays = mapsTo2DArray(results);
   await makeClient();
-  const rows = await writeToSheet(arrays, config.sheetid);
+  const rows = await writeToSheet(arrays, config.sheetid, config.sheetName);
   const maps = rowsToObjects(rows);
-  previousQuads = await objectsToRdf({data: maps}, rml);
-
+  previousData = await objectsToRdf(config, {data: maps}, rml);
   console.log('Synchronisation cold start completed');
 
-  // Pod -> Sheet sync
-  const websocketEndpoints = await getNotificationChannelTypes(config.host + '/.well-known/solid');
-
-  if (websocketEndpoints.length > 0 && websocketEndpoints[0].length > 0 && (!config.noWebsockets)) {
-    // listen using websockets
-    const url = websocketEndpoints[0];
-    const ws = await getWebsocket(url, config.source);
-    ws.on('message', async (notification) => {
-      const content = JSON.parse(notification);
-      if (content.type === 'Update') {
-        const {results} = await queryResource(config, true);
-        const arrays = mapsTo2DArray(results);
-        const maps = rowsToObjects(arrays);
-        const quads = await objectsToRdf({data: maps}, rml);
-        if (!compareArrays(quads, previousQuads, compareQuads)) {
-          const rows = await writeToSheet(arrays, config.sheetid);
-          const maps2 = rowsToObjects(rows);
-          previousQuads = await objectsToRdf({data: maps2}, rml);
-        } else {
-          console.log('got notified but the latest changes are already present');
-        }
-      }
-    });
+  let allOnWebsockets;
+  if (!config.multiple) {
+    allOnWebsockets = await setupResourceListening(config.host, config.source);
   } else {
+    // try to setup a websocket connection for each resource
+    const result = await Promise.all(config.resourceHostmap.map(async (entry) => await setupResourceListening(entry.host, entry.resource)));
+    allOnWebsockets = result.every((e) => e);
+  }
+  if (!allOnWebsockets) {
+    console.log('not all resources supported websockets, polling using timer');
     // polling using timers
     setInterval(async () => {
-      const {results} = await queryResource(config, true);
-      const arrays = mapsTo2DArray(results);
-      const maps = rowsToObjects(arrays);
-      const quads = await objectsToRdf({data: maps}, rml);
-      if (!compareArrays(quads, previousQuads, compareQuads)) {
-        const rows = await writeToSheet(arrays, config.sheetid);
-        const maps2 = rowsToObjects(rows);
-        previousQuads = await objectsToRdf({data: maps2}, rml);
-      }
+      await updateSheet();
```

Comment:
Can't this be replaced with `setInterval(updateSheet)`? Or do we need the explicit `await` for `updateSheet`?